### PR TITLE
feat: support date-fns

### PR DESCRIPTION
**Pull request by fireairforce**
ref issue: https://github.com/ant-design/ant-design/issues/22732

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,6 +80,7 @@
   },
   "peerDependencies": {
     "dayjs": "^1.8.18",
+    "date-fns": "^2.15.0",
```
Comment (afc163) [inline package.json:83]: I think we could ignore `peerDependencies` https://github.com/ant-design/ant-design/issues/23435

Comment (fireairforce) [reply package.json:83]: OK

**Comment by fireairforce** [comment]
The parse function is still in working and i will add test cases after finished `parse`.

**Comment by fireairforce** [comment]
PTAL,thx~

**Comment by zombieJ** [comment]
CI failed. Rebase master.

**Comment by fireairforce** [comment]
All is ok~

**Comment by zombieJ** [comment]
`+ rc-picker@1.14.0`

**Pull request merged**